src/components/ChatEntry.js

```diff
@@ -1,22 +1,46 @@
-import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const onLikeButtonClick = () => {
+    const updateChatEntry = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked,
+    };
+    props.onUpdateLiked(updateChatEntry);
+  };
+
+  const heart = props.liked ? '❤️' : '🤍';
+  const alignment = props.sender === 'Vladimir'? 'chat-entry local' : 'chat-entry remote';
+  const bubbleAlign = props.sender === 'Vladimir' ? 'entry-bubble local' : 'entry-bubble remote';
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
-      <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+    <div className={alignment}>
+      <h2 className="entry-name">{props.sender}</h2>
+      <section className={bubbleAlign}>
+        <p>{props.body}</p>
+        <p className="entry-time">
+          <TimeStamp time={props.timeStamp} />
+        </p>
+        <button className="like" onClick={onLikeButtonClick}>
+          {heart}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  onUpdate: PropTypes.func.isRequired,
```

Comment:
You're receiving this warning due to a typo:

>       Warning: Failed prop type: The prop `onUpdate` is marked as required in `ChatEntry`, but its value is `undefined`.


```suggestion
  onUpdateLiked: PropTypes.func.isRequired,
```